app/services/process_feed_service.rb

```diff
@@ -155,7 +153,8 @@ def status_from_xml(entry)
         reply: thread?(entry),
         language: content_language(entry),
         visibility: visibility_scope(entry),
-        conversation: find_or_create_conversation(entry)
+        conversation: find_or_create_conversation(entry),
+        thread: thread?(entry) ? find_status(thread(entry).first) : nil
```

Comment:
If the remote software does not set “ostatus:conversations”, the result of `find_or_create_conversation` will be `nil` and a new `Conversation` will be created, that's why I added the second line: the validation will use the upstream status' if it exists. As for `find_status`, why wouldn't it be safe to use? It was already in the transaction block. The thing that should not be in the transaction block is `ThreadResolveWorker.perform_async`, which is outside the block.